src/collectors/network_collector.rs

```diff
@@ -0,0 +1,22 @@
+use network_interface::NetworkInterface;
+use network_interface::NetworkInterfaceConfig;
+
+pub struct NetworkInformation {
+    name: String,
+    addr: String,
+    netmask: String,
+
+}
+
+
+pub fn collect_network_information() -> Vec<NetworkInterface> {
+    /*
+     * Collect information about all available network interfaces.
+     *
+     * Will return a Vector containing a JSON-like data format.
+     */
+    let network_interfaces: Vec<NetworkInterface> = NetworkInterface::show().unwrap();
```

Comment:
This code is currently not using the struct defined above

Reply:
You are right. I'm thinking of deconstructing the Vector I get from the `NetworkInterface` crate and build `NetworkInformation` objects for each of the `NetworkInterface` objects within it containing all its fields. But I'll have to take a deeper look in how that would work.

Reply:
For now, I will move the network information collection to a different PR as this will become quite the act.